shortfin/tests/invocation/mobilenet_tracing_test.py

```diff
@@ -0,0 +1,251 @@
+import signal
+import sys
+import time
+from pathlib import Path
+
+import socket
+from contextlib import closing
+
+
+def find_free_port():
+    with closing(socket.socket(socket.AF_INET, socket.SOCK_STREAM)) as s:
+        s.bind(("", 0))
+        s.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+        return s.getsockname()[1]
+
+
+logger = logging.getLogger(__name__)
+
+# skip if iree-tracy-capture is not in $PATH
+def is_tracy_capture_available():
+    """Check if iree-tracy-capture is available on the system."""
+    return shutil.which("iree-tracy-capture") is not None
+
+
+# Check if shortfin was built with tracing enabled
+def is_tracing_enabled():
+    """Check if shortfin was built with tracing enabled."""
+    import importlib
+
+    try:
+        importlib.import_module("_shortfin_tracy")
+        return True
+    except ModuleNotFoundError:
+        # Only logged at debug level since we'll have a warning later
+        logger.debug("_shortfin_tracy module not found")
+        return False
+
+
+def run_mobilenet(model_path):
+    """Run MobileNet model with tracing enabled.
+
+    Args:
+        model_path: Path to the compiled VMFB file
+    """
+    # Import shortfin here to avoid connecting to tracy at module import time
+    import shortfin as sf
+    import shortfin.array as sfnp
+
+    # Check environment variable for tracing
+    runtime = os.environ.get("SHORTFIN_PY_RUNTIME", "")
+    print(f"-- Using {runtime} runtime (SHORTFIN_PY_RUNTIME={runtime})")
+
+    logging.basicConfig(level=logging.INFO)
+    logger.info(f"Running MobileNet model from {model_path}")
+
+    # Create system and fiber with CPU backend
+    sc = sf.host.CPUSystemBuilder()
+    system = sc.create_system()
+    fiber = system.create_fiber()
+    device = fiber.device(0)
+
+    # Load model
+    program_module = system.load_module(model_path)
+    program = sf.Program([program_module], devices=system.devices)
+    main_function = program["module.torch-jit-export"]
+
+    # Create input tensor
+    dummy_data = array.array(
+        "f", ([0.2] * (224 * 224)) + ([0.4] * (224 * 224)) + ([-0.2] * (224 * 224))
+    )
+    device_input = sfnp.device_array(device, [1, 3, 224, 224], sfnp.float32)
+    staging_input = device_input.for_transfer()
+    with staging_input.map(discard=True) as m:
+        m.fill(dummy_data)
+    device_input.copy_from(staging_input)
+
+    # Run model
+    async def run_model():
+        logger.info("Running MobileNet inference")
+        (device_output,) = await main_function(device_input, fiber=fiber)
+        # Transfer output back to host
+        host_output = device_output.for_transfer()
+        host_output.copy_from(device_output)
+        await device
+        # Clean up
+        del device_output
+        del host_output
+        logger.info("MobileNet inference completed")
+
+    system.run(run_model())
+
+    # Clean up
+    system.shutdown()
+    logger.info("MobileNet model execution finished")
```

Comment:
I wonder if we can hide some of this system setup code behind a nicer API, especially if we add other integration tests like this one that just want "some server running a model for one request".

Reply:
One thing that keeps coming up with the LLAMA models is ...

"Oh no! we have a problem!"
"Does it replicate on iree-run-module?"
"Ug it's a lot of work to get the mlir and the inputs and the flags"

Now if we had an interface like that, that also, in an optional debug mode, dumps the inputs and the flags for iree-run-module....

Reply:
Specifically I'm thinking about something where

Let's say we have a program function fn in a shortfin_app

and we have an `await fn(args)`

By setting an enviornmental variable

We can make it dump all the inputs & the flags / options used to set up the system and the fibers and devices.

----

Once we have the dump, we can use an interface `shortfin-run-module` to run it, and that'd be the `some server running a model for 1 request`

Or just test with `iree-run-module` to make sure that the problem isn't just a shortfin problem.

Reply:
A `shortfin-run-module` interface would be pretty neat.

Would also make it easier to test discrepancies between sf and iree run module